### DISCUSSION TITLE
How to use downloaded packages in satis.json?

### DISCUSSION THREAD

**Discussion by sampleshub** [composer/satis#702] [Q&A]
For my testing of Satis so far, I've been using links to GitHub repos:

```
{
    "name": "My Private Repository",
    "homepage": "http://localhost/",
    "repositories": [
        {
            "name": "doctrine/inflector",
            "type": "git",
            "url": "https://github.com/doctrine/inflector.git"
        },
        {
            "name": "laravel/laravel",            
            "type": "git",
            "url": "https://github.com/laravel/laravel.git"
        }
        
    ],
    "require-all": true,
    "archive": {
        "directory": "dist"
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true,
        "secure-http": false
    }
}
```

But now I want only specific version as a package, not the entire repository with all versions.

So let's say I only download the latest version as zip file from GitHub. How can I add that to the above code so that when I run `php bin/satis build` I do not end up with errors? 

All attempts resulted in all sorts of errors as I couldn't find the correct syntax.
I already created a folder containing all the downloaded zip files

**Comment by SvenRtbg** [comment]
Don't use `require-all:true`, but use `require:{"vendor1/package1":">=2.1.3",...}` just like composer would do. Remember to use open-ended constraints if you want to allow major update fetches, and consider that Satis may add lower versions if other packages require them.

**Reply by sampleshub** [reply]
Your solution would work in case I want to install specific packages from online repositories, but in my case I need to be able to setup my local repository from local zip files, not from GitHub. (To be able to use the `php bin/satis build` on zip files from my local machine)

I managed to get it working with the following example `satis.config` file:

```
{
    "name": "local/repository",
    "homepage": "http://localhost/",
    "repositories": [            
        {
            "type": "package",
            "package": {
                "name": "some_vendor/package_name",
                "version": "2.0.4",
                "dist": {
                    "url": "C:/path/to/zip_files/package_name.zip",
                    "type": "zip"
                }
            }
        }       
    ],
    "require-all": true,
    "archive": {
        "directory": "dist"
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true,
        "secure-http": false
    }
}
```
